src/Simplex/FileTransfer/Agent.hs

```diff
@@ -287,17 +294,27 @@ runXFTPRcvLocalWorker c Worker {doWork} = do
           throwError $ INTERNAL "no chunk path"
 
 xftpDeleteRcvFile' :: AgentMonad m => AgentClient -> RcvFileId -> m ()
-xftpDeleteRcvFile' c rcvFileEntityId = do
-  rcvFile@RcvFile {rcvFileId} <- withStore c $ \db -> getRcvFileByEntityId db rcvFileEntityId
-  handleError (const $ pure ()) $ withStore' c (`getRcvFileRedirects` rcvFileId) >>= mapM_ remove
-  remove rcvFile
+xftpDeleteRcvFile' c rcvFileEntityId = xftpDeleteRcvFiles' c [rcvFileEntityId]
+
+xftpDeleteRcvFiles' :: AgentMonad m => AgentClient -> [RcvFileId] -> m ()
+xftpDeleteRcvFiles' c rcvFileEntityIds = do
+  (_, rcvFiles) <- partitionEithers <$> withStoreBatch c (\db -> map (getById db) rcvFileEntityIds)
+  (_, redirects) <- partitionEithers <$> withStoreBatch' c (\db -> map (getRedirects db) rcvFiles)
+  let rcvFilesAll = concat redirects <> rcvFiles
+      pathsToRemove = filter (\RcvFile {status} -> status == RFSComplete || status == RFSError) rcvFilesAll
+  forM_ pathsToRemove $ \RcvFile {prefixPath} -> removePath prefixPath `catchAgentError` const (pure ())
```

Comment:
why we only delete some files from storage and then still delete all of them from the database?